src/app.ts

```diff
@@ -491,73 +515,49 @@ export class App {
           if (method.description) {
             out.comment(formatComment(method.description));
           }
-          const requestParameters: Record<
-            string,
-            gapi.client.discovery.JsonSchema
-          > = sortObject({ ...parameters, ...method.parameters });
-          const hasRequestRef = method.request && method.request['$ref'];
-          if (
-            !(requestParameters.hasOwnProperty('resource') && hasRequestRef)
-          ) {
-            // no resource param and no body at the same time -> generate x(request)
+
+          const requestRef = method.request?.$ref;
+          const requestParameters: Record<string, JsonSchema> = sortObject({
+            ...parameters,
+            ...method.parameters,
+          });
+
+          if (!requestParameters.resource || !requestRef) {
+            // generate method(request)
             out.method(
               formatPropertyName(checkExists(getName(method.id))),
               [
                 {
                   parameter: 'request',
-                  type: (writer: TypescriptTextWriter) => {
-                    writer.anonymousType(() => {
-                      _.forEach(requestParameters, (data, key) => {
-                        if (data.description) {
-                          writer.comment(formatComment(data.description));
-                        }
-                        writer.property(
-                          key,
-                          getType(data, schemas),
-                          data.required || false
-                        );
-                      });
-
-                      if (method.request && method.request['$ref']) {
-                        writer.comment('Request body');
-                        writer.property(
-                          'resource',
-                          method.request['$ref'],
-                          true
-                        );
-                      }
-                    });
-                  },
+                  type: App.createRequestParameterWriterCallback(
+                    requestParameters,
+                    schemas,
+                    requestRef
+                  ),
+                  required: Boolean(requestRef),
```

Comment:
This was one of the primary changes: The `request` argument should be optional if there is no (required) request body.